### PR TITLE
Lint add warnings instead of check failure

### PR DESCRIPTION
**Pull request by Snorch**
There are multiple cases where good human readable code block is
converted to an unreadable mess by clang-format, so we don't want to
rely on clang-format completely. Also there is no way, as far as I can
see, to make clang-format only fix what we want it to fix without
breaking something.

So let's just display hints inline where clang-format is unhappy. When
reviewer sees such a warning it's a good sign that something is broken
in codding style around this warning.

### CODE REVIEW

**Comment by Snorch** [comment]
It looks like that:

image

Please remove https://github.com/checkpoint-restore/criu/pull/2250/commits/584515db3903ded398da012f781b4b1e93fcd344 when merging this PR, this commit is needed just to illustrate the concept.

File: criu/mount.c [hunk 1]
```diff
@@ -95,12 +95,12 @@ int ext_mount_parse_auto(char *key)
 static char *ext_mount_lookup(char *key)
 {
 	char *v;
-	int len = strlen(key);
+int len = strlen(key);
```
Comment (avagin) [inline criu/mount.c:98]: > clang-format: Possible codding-style problem

can it give more info about a problem?

Comment (Snorch) [reply criu/mount.c:98]: I feel that hiding clang-generated auto-formats is a good thing, else we would end up blindly applying them again and code readability would suffer 😀 It should be enough to point the line in file to make manual review a bit easier.

Moreover, multiline warnings seem to be not working on github https://github.com/actions/toolkit/issues/193#issuecomment-1691046058 . And also there is no way to get info from clang about type of error.

File: scripts/github-indent-warnings.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                file_name = file_matches[0]
+                continue
+
+            if file_name is None:
+                continue
+
+            line_matches = re.findall(re_line, line)
+            if len(line_matches) == 1:
+                line_number = int(line_matches[0]) + 3
+                print(f'::warning file={file_name},line={line_number}::clang-format: Possible codding-style problem')
```
Comment (rst0git) [inline scripts/github-indent-warnings.py:28]:
```Suggestion
                print(f'::warning file={file_name},line={line_number}::clang-format: Possible coding-style problem')
```

File: scripts/github-indent-warnings.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+re_file = r'^diff --git a/(\S\S*)\s.*$'
+re_line = r'^@@ -(\d\d*)\D.*@@.*$'
+
+if __name__ == '__main__':
+    if len(sys.argv) != 1 and len(sys.argv) != 2:
+        print(f'usage: {sys.argv[0]} <path/to/file>')
+        print(f'usage: <command> | {sys.argv[0]}')
+        exit(1)
+
+    with open(sys.stdin.fileno() if len(sys.argv) == 1 else sys.argv[1], 'r') as fi:
```
Comment (rst0git) [inline scripts/github-indent-warnings.py:13]: Would it be possible to split this into multiple lines? For example, something like the following:
```Python
input_file = sys.stdin.fileno()
if len(sys.argv) == 1:
    input_file = sys.argv[1]

with open(input_file, 'r') as fi:
   ...
```
([*simple is better than complex*](https://peps.python.org/pep-0020/))

File: scripts/github-indent-warnings.py (hunk 2 above)
Comment (Snorch) [reply scripts/github-indent-warnings.py]: Thanks, done!

File: scripts/github-indent-warnings.py (hunk 3 above)
Comment (Snorch) [reply scripts/github-indent-warnings.py]: Sure, you are right, done.

**Comment by Snorch** [comment]
I've added changes to CONTRIBUTING.md with bad clang-formating examples to PR and a link from warnings to CONTRIBUTING.md, so that it would be easier to understand how this warning should be handled.

**Comment by Snorch** [comment]
I've removed test commit to trigger warnings (it works), so it's now ready to merge.

**Pull request merged**